### PR TITLE
[stm32] i2c irq priority can be set

### PR DESCRIPTION
**Pull request by victorandrehc**
Add the possibility to set the i2c irq priority.

### CODE REVIEW

File: src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in [hunk 1]
```diff
@@ -99,12 +99,12 @@ public:
 	 */
 	template<class SystemClock, baudrate_t baudrate=kBd(100), percent_t tolerance=pct(5)>
 	static void
-	initialize()
+	initialize(uint32_t isrPriority = 10u)
```
Comment (klsc-zeat) [inline src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in]: i have a feeling the modm guys probably want the priority as a template parameter -> how is it handled on CAN for example?

**Review by klsc-zeat — APPROVED** [review]

File: src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in (hunk 1 above)
Comment (victorandrehc) [reply src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in]: I will check that. Those are just the commits we made on our local branch as is. What is bothering me the most for now is why the unit tests are failing somewhere we have nothing to do with.

Comment (victorandrehc) [reply src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in]: it is a parameter passed on constructor. Honestly I would keep as is then.
`modm::platform::Can{{ id }}::initializeWithPrescaler(
		uint16_t prescaler, uint8_t bs1, uint8_t bs2,
		uint32_t interruptPriority, Mode startupMode, bool overwriteOnOverrun)`

Comment (victorandrehc) [reply src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in]: They also don't have a constructor for the I2c, honestly the best solution is to leave as is

Comment (klsc-zeat) [reply src/modm/platform/i2c/stm32-extended/i2c_master.hpp.in]: ok if they accept it as is, it is fine for me! ;)

**Comment by victorandrehc** [comment]
PR merged on modm upstream

**Pull request closed without merging**